### PR TITLE
feat-v2(front_validation): Adding front validation using symfony

### PR DESCRIPTION
**Pull request by Ngob**
Adding management of symfony violation in frontend

### CODE REVIEW

File: apps/back/src/Controller/DemoController.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+use App\DevTools\PHPStan\ThisRouteDoesntNeedAVoter;
+use App\Dto\Request\Demo\CreateDemoValidationDto;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpKernel\Attribute\MapRequestPayload;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Http\Attribute\IsGranted;
+
+class DemoController extends AbstractController
```
Comment (nguyenk) [inline apps/back/src/Controller/DemoController.php]: Maybe we can give a more understandable name to the Controller, like `ValidationDemoController` ?

File: apps/back/src/Controller/UserController.php [hunk 2]
```diff
@@ -31,9 +30,11 @@ public function __construct(
     }
 
     #[Route('/users', name: 'create_user', methods: ['POST'])]
+    #[IsGranted('IS_AUTHENTICATED_FULLY')]
```
Comment (nguyenk) [inline apps/back/src/Controller/UserController.php]: What's the point of having both checks : Voter + IS_AUTHENTICATED_FULLY ?

File: apps/back/src/Entity/User.php [hunk 3]
```diff
@@ -12,24 +12,22 @@
...
-    #[ORM\GeneratedValue]
-    #[ORM\Column]
-    private int|null $id = null;
-
     #[ORM\Column(length: 180)]
     private string $password;
 
     /** @param array<string> $roles */
     public function __construct(
+        #[ORM\Id]
```
Comment (nguyenk) [inline apps/back/src/Entity/User.php:20]: Didn't we decided to go for public props rather then `consctruct` for DTOs ?

**Review by nguyenk — COMMENTED** [review]
All good, very small feedbacks

File: apps/back/src/Entity/User.php (hunk 3 above)
Comment (Ngob) [reply apps/back/src/Entity/User.php:20]: but its an entity

**Pull request merged**